pkg/etcd/etcd.go

```diff
@@ -1162,9 +1202,7 @@ func (e *ETCD) trackLearnerProgress(ctx context.Context, progress *learnerProgre
 	return e.setLearnerProgress(ctx, progress)
 }
 
-func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client corev1.CoreV1Interface, memberName string, promoted bool) (*v1.Node, []byte, error) {
-	etcdStatusType := v1.NodeConditionType("EtcdIsVoter")
-
+func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client kubernetes.Interface, memberName string, promoted bool, isMember bool, isHealthy bool) error {
```

Comment:
We've picked up a lot of bool flags here. Rather than try to track the various combinations, perhaps we should just create a `MemberStatus` enum to track the possible states?

```mermaid
stateDiagram
  unjoined --> learner
  learner --> voter
  learner --> unjoined
  voter --> unhealthy
  unhealthy --> voter
  unhealthy --> unjoined
  
```

You'll have to adjust the callers to track these changes.

```suggestion
func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client kubernetes.Interface, name string, status MemberStatus) error {
```